src/test/java/discovery/controller/DiscoveryControllerTestMockito.java

```diff
@@ -0,0 +1,188 @@
+package discovery.controller;
+
+
+import discovery.model.Discover;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import org.springframework.http.MediaType;
+
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+
+
+import java.util.concurrent.ConcurrentHashMap;
+
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+
+
+
+public class DiscoveryControllerTestMockito {
+
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private DiscoveryController discoveryController;
+
+    @Mock
+    private ConcurrentHashMap<String, Discover> concurrentHashMap;
+
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        this.mockMvc = MockMvcBuilders.standaloneSetup(discoveryController).build();
+    }
+
+    @Test
+    public void shouldGetExistingService() throws Exception {
+        //given
+        String key="lazylogin";
+        String value="http://lazylogin.trafalgar.ws/";
+
+        //when
+        when(concurrentHashMap.get(key)).thenReturn(new Discover(key,value));
+
+        //then
+        mockMvc.perform(get("/discover/lazylogin")).andExpect(status().isOk());
+    }
+
+    @Test
+    public void shouldReturnNotFoundWhenGettingService() throws Exception {
+        //given
+        String key="lazylogin2";
+        String value="http://lazylogin2.trafalgar.ws/";
+
+        //when
+        when(concurrentHashMap.get(key)).thenReturn(new Discover(key,value));
+
+        //then
+        mockMvc.perform(get("/discover/lazylogin")).andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void shouldCreateNewService() throws Exception {
```

Comment:
I've changed to make it clear. The idea is you have your concurrenthashmap with lazylogin kel-value and if you ask for lazylogin2 the program will not find it